lib/build/ProjectBuilder.js

```diff
@@ -30,6 +30,67 @@ class ProjectBuilder {
 	 * 			If the wildcard '*' is provided, only the included tasks will be executed.
 	 */
 
+	/**
+	 * Alternatively to providing plain lists of dependency names to include and exclude, a more complex
+	 * "Dependency Includes" object can be provided to define which dependencies should be part of the build result.
+	 * <br>
+	 * The provided information will be used to compile lists of <code>includedDependencies</code> and
+	 * <code>excludedDependencies</code>, which will be used during the build process.
+	 * <br><br>
+	 * Regular expression-parameters are directly applied to a list of all project dependencies
+	 * so that they don't need to be evaluated in later processing steps.
+	 * <br><br>
+	 * Generally, includes are handled with a higher priority than excludes. Additionally, operations for processing
+	 * transitive dependencies are handled with a lower priority than explicitly mentioned dependencies. The "default"
+	 * dependency-includes are appended in the end.
+	 * <br><br>
+	 * The priority of the various dependency lists is applied in the following order.
+	 * But note that a later exclude can't overrule an earlier include:
+	 * <br>
+	 * <ol>
+	 *   <li><code>includeDependency</code>, <code>includeDependencyRegExp</code></li>
+	 *   <li><code>excludeDependency</code>, <code>excludeDependencyRegExp</code></li>
+	 *   <li><code>includeDependencyTree</code></li>
+	 *   <li><code>excludeDependencyTree</code></li>
+	 *   <li><code>defaultIncludeDependency</code>, <code>defaultIncludeDependencyRegExp</code>,
+	 *     <code>defaultIncludeDependencyTree</code></li>
+	 * </ol>
+	 *
+	 * @public
+	 * @typedef {object} @ui5/project/build/ProjectBuilder~DependencyIncludes
+	 * @property {object} parameters Complex dependency includes
```

Comment:
Why is there a `parameters` property here?

Reply:
Done